pkg/scalers/kafka_scaler.go

```diff
@@ -121,9 +122,29 @@ func NewKafkaScaler(config *ScalerConfig) (Scaler, error) {
 
 func parseKafkaAuthParams(config *ScalerConfig, meta *kafkaMetadata) error {
 	meta.saslType = KafkaSASLTypeNone
-	if val, ok := config.AuthParams["sasl"]; ok {
-		val = strings.TrimSpace(val)
-		mode := kafkaSaslType(val)
+	saslMetadata := config.TriggerMetadata["sasl"]
+	saslAuthParam := config.AuthParams["sasl"]
+	tlsMetadata := config.TriggerMetadata["tls"]
+	tlsAuthParam := config.AuthParams["tls"]
+
+	if (!reflect.ValueOf(saslMetadata).IsZero() && !reflect.ValueOf(saslAuthParam).IsZero()) || (!reflect.ValueOf(tlsMetadata).IsZero() && !reflect.ValueOf(tlsAuthParam).IsZero()) {
+		return errors.New("cannot set `sasl` or `tls` values in both metadata and spec.secretTargetRef")
+	}
+
+	if (!reflect.ValueOf(saslMetadata).IsZero() && !reflect.ValueOf(tlsAuthParam).IsZero()) || (!reflect.ValueOf(saslAuthParam).IsZero() && !reflect.ValueOf(tlsMetadata).IsZero()) {
+		return errors.New("both`sasl` and `tls` must be in either metadata or spec.secretTargetRef")
+	}
+
+	var sasl string
+	if !reflect.ValueOf(saslMetadata).IsZero() {
+		sasl = saslMetadata
+	} else if !reflect.ValueOf(saslAuthParam).IsZero() {
+		sasl = saslAuthParam
+	}
+
+	if !reflect.ValueOf(sasl).IsZero() {
+		sasl = strings.TrimSpace(sasl)
+		mode := kafkaSaslType(sasl)
```

Comment:
There is a [helper function](https://github.com/kedacore/keda/blob/main/pkg/scalers/scaler.go#L114) that I guess can simplify this code, PTAL